csharp/extractor/Semmle.Extraction.CSharp/SymbolExtensions.cs

```diff
@@ -224,48 +272,90 @@ static void BuildNamedTypeId(this INamedTypeSymbol named, Context cx, TextWriter
                     {
                         trapFile.Write(f.Name);
                         trapFile.Write(":");
-                        subTermAction(cx, tb0, f.Type);
+                        f.Type.BuildOrWriteId(cx, tb0, symbolBeingDefined, addBaseClass);
                     }
                     );
                 trapFile.Write(")");
                 return;
             }
 
-            if (named.ContainingType != null)
-            {
-                subTermAction(cx, trapFile, named.ContainingType);
-                trapFile.Write('.');
-            }
-            else if (named.ContainingNamespace != null)
+            void AddContaining()
             {
-                if (prefixAssembly)
-                    BuildAssembly(named.ContainingAssembly, trapFile);
-                named.ContainingNamespace.BuildNamespace(cx, trapFile);
+                if (named.ContainingType != null)
+                {
+                    named.ContainingType.BuildOrWriteId(cx, trapFile, symbolBeingDefined, addBaseClass);
+                    trapFile.Write('.');
+                }
+                else if (named.ContainingNamespace != null)
+                {
+                    if (cx.AddAssemblyTrapPrefix && !(named.ContainingAssembly is null))
```

Comment:
`!(X is null)` can be "shortened" to `X is object`.

Reply:
Looks like C# 9 will support `X is not null`; lets use `X is object` for now.